src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ResolutePatch.cs

```diff
@@ -0,0 +1,142 @@
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+    
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+      
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, 50f,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        _perk.IncrementType
+      );
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, postfix: new HarmonyMethod(PatchMethodInfoPostfix));
+      Applied = true;
+    }
+    
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void Postfix(ref float __result, MobileParty mobileParty, StatExplainer explanation = null) {
+      var perk = ActivePatch._perk;
+      
+      if (mobileParty.SiegeEvent == null) return;
+      var partyMemberBonus = GetPartyMemberBonus(mobileParty, perk);
+      if (partyMemberBonus < .01) return;
+
+      var partyMembersCount = (int) (partyMemberBonus / perk.PrimaryBonus);
+      var multiplicativeBonus = CalculateMultiplicativeBonus(perk.PrimaryBonus / 100, partyMembersCount);
+      var moraleLossReductionRate = multiplicativeBonus * -1;
+      
+      var totalMoraleLossReduction = CalculateTotalMoraleLossReduction(mobileParty, moraleLossReductionRate);
+      explanation?.AddLine(perk.Name.ToString(), totalMoraleLossReduction);
+      __result += totalMoraleLossReduction;
+    }
+    
+    private static float GetPartyMemberBonus(MobileParty mobileParty, PerkObject perk)
+    {
+      var perkPartyMemberValue = new ExplainedNumber(100);
+      PerkHelper.AddPerkBonusForParty(perk, mobileParty, ref perkPartyMemberValue);
+      var additiveMoralePenaltyReductionRate = perkPartyMemberValue.ResultNumber - perkPartyMemberValue.BaseNumber;
+      return additiveMoralePenaltyReductionRate;
+    }
+
+    private static float CalculateMultiplicativeBonus(float partyMemberBonus, int partyMemberCount) {
+      var finalRate = 0f;
+      var lastRate = 1f;
+      
+      for (var i = 0; i < partyMemberCount; i++) {
+        var nextRate = partyMemberBonus * lastRate;
+        finalRate += nextRate;
+        lastRate = nextRate;
+      }
+
+      return finalRate;
+    }
+
+    private static int CalculateTotalMoraleLossReduction(MobileParty mobileParty, float moraleLossReductionRate)
+    {
+      var recentEventsMoraleLossReduction = Math.Min(mobileParty.RecentEventsMorale, 0) * moraleLossReductionRate;
+      var starvationMoraleLossReduction = mobileParty.Party.IsStarving ? -30 * moraleLossReductionRate : 0;
+      var unpaidWageMoraleLossReduction = mobileParty.HasUnpaidWages > 0f ? -20 * moraleLossReductionRate : 0;
```

Comment:
Use the existing `GetStarvationMoralePenalty` and `GetNoWageMoralePenalty` functions instead of hardcoded numbers

Reply:
I'll have it done